library/core/functions.general.php

```diff
@@ -214,6 +214,26 @@ function arrayValueI($Needle, $Haystack, $Default = false) {
     }
 }
 
+if (!function_exists('arraySelect')) {
+    /**
+     * Select the first non-empty value from an array.
+     *
+     * @param array $keys An array of keys to try.
+     * @param array $array The array to select from.
+     * @param mixed $default The default value if non of the keys exist.
+     * @return mixed Returns the first non-empty value of {@link $default} if none are found.
+     * @category Array Functions
+     */
+    function arraySelect(array $keys, array $array, $default = null) {
```

Comment:
Sorry about this, but can we make this a private method of Schema instead? I want to reduce dependencies as much as possible.